storagenode/storagenodedb/orders.go

```diff
@@ -146,10 +146,36 @@ func (db *ordersdb) ListUnsentBySatellite(ctx context.Context) (_ map[storj.Node
 }
 
 // Archive marks order as being handled.
-func (db *ordersdb) Archive(ctx context.Context, satellite storj.NodeID, serial storj.SerialNumber, status orders.Status) (err error) {
+func (db *ordersdb) Archive(ctx context.Context, requests ...orders.ArchiveRequest) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	result, err := db.db.Exec(`
+	txn, err := db.Begin()
+	if err != nil {
+		return ErrInfo.Wrap(err)
+	}
+	defer func() {
+		if err == nil {
+			err = txn.Commit()
+		} else {
+			err = errs.Combine(err, txn.Rollback())
+		}
+	}()
+
+	for _, req := range requests {
+		err := db.archiveOne(ctx, txn, req)
```

Comment:
Have you considered putting all the necessary statements together in one longish query? There isn't any real need for an explicit transaction block here, and this look like an awful lot of back-and-forth time. That might not matter-  I don't think anything much cares whether this is fast- but it could reduce database load.

The query could just be the concatenation of all the queries from archiveOne, or that query could be modified to have a clause like

    WHERE (satellite_id = ? AND serial_number = ANY(?, ?, ?, ?, ...))
        OR (satellite_id = ? AND serial_number = ANY(?, ?, ?, ...))

Reply:
sorry, clarification- that would only work on postgresql. I imagine this needs to work against sqlite still.

Reply:
There are probably better ways to accomplish this, but I just want to do the easiest thing that stops doing a full transaction/fsync per order. I also don't know if sqlite's query engine is capable enough to handle one huge query cheaper than many small queries.

Also, I think it only has to work against sqlite.

Reply:
Oh right this is storagenode. Oops. Then yeah, the extra transaction block and round trip time are non-issues. Sorry for noise